extensions/metrics/src/observers/pushgateway.observer.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/extension-metrics
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {config, LifeCycleObserver} from '@loopback/core';
+import {Pushgateway} from 'prom-client';
+import {MetricsBindings} from '../keys';
+import {DEFAULT_METRICS_OPTIONS, MetricsOptions} from '../types';
+
+export class PushGatewayObserver implements LifeCycleObserver {
```

Comment:
I find this name weird. To me, it implies this class is observing a push gateway. Please find a better one, also rename the file accordingly.

It's a good practice to use names that express the intent (what is this trying to achieve) instead of implementation details (it's implemented as a LifeCycleObserver).

IIUC, this code is setting up periodic reporting of metrics to a push gateway. So maybe `MetricPushReporter` would be a good name?

```suggestion
export class MetricPushReporter implements LifeCycleObserver {
```